docs/docs/development.md

```diff
@@ -0,0 +1,169 @@
+# Testing
+
+## Local Testing
+
+Local testing is a great way to test the components of QHub. While it
+does test most of everything it does not test cloud provisioned
+components such as the managed kubernetes cluster, vpcs, managed
+container registries, etc.
+
+This guide assumes that you have the QHub Cloud repository downloaded
+and you are at the root of the repository.
+
+### Dependencies
+
+In order to develop with QHub you will need to download the following
+dependencies and have them available in your path.
+
+ - [terraform](https://www.terraform.io/downloads.html)
+ - [minikube](https://v1-18.docs.kubernetes.io/docs/tasks/tools/install-minikube/)
+
+### Initialize kubernetes cluster
+
+Testing is done with minikube. Note that this will download a ~500MB
+docker image.
+
+```shell
+minikube start --cpus 2 --memory 4096 --driver=docker
+```
+
+The jupyterlab instances require mounting nfs pvcs. This requires
+nfs-common drivers be installed on the nodes themselves.
+
+```shell
+minikube ssh "sudo apt update; sudo apt install nfs-common -y"
+```
+
+Configure the `metallb` load balancer to have a start ip of
+`172.17.10.100` and an end ip of `172.17.10.200`. These ips were not
+randomly chosen. You must make sure that the ip range is within the
```

Comment:
> These ips were not randomly chosen.

I think it would be good to document, the motivation behind choosing this IP Range, since the doc says it was not randomly chosen, but doesn't mentions the reason behind it.